### ISSUE TITLE
Fix Condition model checking verify from segment data from the API not from the model code

### ISSUE THREAD

**Issue by zachaysan** [SolidStateGroup/bullet-train-api#4782]

**Comment by matthewelwell** [comment]
@zachaysan can you add a description here (and maybe reword the title of the issue)? I have no idea what this issue is relating to.

**Comment by zachaysan** [comment]
Ok I've reworded the title but it turns out that this issue was either due to a misconception or possibly due to now-deleted code. I've verified that the segment enforcement is already at the API layer, not in the model code so I'm going to close this issue.

Also, the linked Sentry issue looked related, but is in fact unrelated. I disconnected this issue from the sentry alert on Sentry's end but the change didn't propagate to this issue for some reason.
